base/linalg/diagonal.jl

```diff
@@ -101,9 +101,9 @@ end
 parent(D::Diagonal) = D.diag
 
 ishermitian{T<:Real}(D::Diagonal{T}) = true
-ishermitian(D::Diagonal) = all(D.diag .== real(D.diag))
+ishermitian(D::Diagonal) = isreal(D.diag)
```

Comment:
Discussion for another time and place, but there's sometimes a tension between mathematical completeness and software pragmatism, e.g. getting a vector in a function that was written assuming scalar inputs and not noticing until a ways downstream.